s3-uploader/runtimes/cpp11/Dockerfile

```diff
@@ -1,26 +1,36 @@
-FROM public.ecr.aws/lambda/provided:al2 as builder
+ARG ARCH
+FROM public.ecr.aws/lambda/provided:al2-$ARCH AS builder
 ENV APP_BASE_DIR=/tmp/app
 WORKDIR $APP_BASE_DIR
-RUN yum install cmake3 make g++ git zip libcurl-devel zlib-devel openssl-devel -y
+RUN yum install cmake3 git zip autoconf automake -y
 RUN yum groupinstall "Development Tools" -y
 
 COPY . $APP_BASE_DIR
 WORKDIR $APP_BASE_DIR
 SHELL ["/bin/bash", "-c"]
+# Build static libcurl from source
+RUN curl -O https://curl.se/download/curl-7.88.1.tar.gz
+RUN tar -xvf curl-7.88.1.tar.gz
+WORKDIR "${APP_BASE_DIR}/curl-7.88.1"
+RUN autoreconf -fi
+RUN ./configure --disable-shared --disable-pthreads --disable-threaded-resolver \
+--with-pic --without-ssl --without-zlib --prefix=${APP_BASE_DIR}/lambda-install
+RUN make -j && make install
+# Build the C++ Lambda runtime
+WORKDIR "${APP_BASE_DIR}"
 RUN git clone https://github.com/awslabs/aws-lambda-cpp.git
 WORKDIR "${APP_BASE_DIR}/aws-lambda-cpp/"
 RUN mkdir build
 WORKDIR "${APP_BASE_DIR}/aws-lambda-cpp/build"
 RUN cmake3 .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=~/lambda-install
```

Comment:
Did you try building the image? It seems to fail on my computer with the following error:
```
 cmake3 .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=~/lambda-install
-- The CXX compiler identification is GNU 7.3.1
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /usr/share/cmake3/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake3/Modules/FindPackageHandleStandardArgs.cmake:445 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake3/Modules/FindCURL.cmake:169 (find_package_handle_standard_args)
  CMakeLists.txt:41 (find_package)


-- Configuring incomplete, errors occurred!
```

Reply:
Ok I've found the issue! (see other comment on the prefix)

Reply:
I definitely built it a few times, but maybe ended up pushing something that didn't have the latest changes. Apologies.

Reply:
no worries!